packages/runtime/runtime-definitions/src/aliasing.md

```diff
@@ -0,0 +1,48 @@
+# Aliasing and root datastores
+
+## What is a root datastore? What is an aliased datastore?
+
+A root datastore is a datastore which is addressable by a custom identifier (client supplied) and is a direct child of the container runtime. A root datastore will never be garbage collected and it is a singleton with regards to the runtime it belongs to. The concept of a root datastore is considered legacy, mainly due to the previous (deprecated and removed) APIs which referred to such datastores as root. 'Aliased datastore' is the preferred current term to refer to such datastores.
+
+## When to alias datastores?
+
+Alias needs to happen for datastores which:
+
+-   must be referenced by a custom id
+-   must never be garbage collected (as they are bound to a custom id which may be stored by the client, they must always be available to be referenced)
+-   must be singletons
+
+The legacy way of creating root datastores was vulnerable to name conflicts, as two clients attempting to create the same root datastore with the same id risks corrupting the document. Aliasing changed the way to achieve the same goal, by enabling an asynchronous 'aliasing' operation on any newly created datastore. So in order to create such a datastore, the client needs to create an anonymous datastore (which will receive a newly generated UUID) and then explicitly attempt to bind it to a custom id (the alias) within a different operation which is decoupled from its creation.
```

Comment:
I'd not talk about legacy. That said, I'd have a section on - options when a data store is not a singleton, i.e. it's referenced through handle from some DDS. But is created on demand (and thus this on-demand creation creates an avenue for race conditions). And mention that doing it using non-consensus-based DDSs (like map, directory) will not work, but using something like ConsensusRegistry will allow correctness.

Alternatively, clients could use aliasing (by creating a naming space, like maybe in the form of {parent_name}_{suffix}, but it has a huge con - it will not be collectable!
This is actually interesting to pursue - should we allow decoupling of aliasing and rootiness to help with this scenario?

Reply:
I feel like this level of scenario detail is unrelated to the aliasing feature, which is what this readme is targeting.